apps/etterlatte-saksbehandling-ui/client/src/components/aktivitetsplikt/brev/FerdigstillModal.tsx

```diff
@@ -0,0 +1,97 @@
+import React, { useState } from 'react'
+import { Alert, BodyShort, Button, Heading, Modal, VStack } from '@navikt/ds-react'
+import { setAktivitetspliktOppgave } from '~store/reducers/Aktivitetsplikt12mnd'
+import { useApiCall } from '~shared/hooks/useApiCall'
+import { ferdigstillBrevOgOppgaveAktivitetsplikt } from '~shared/api/aktivitetsplikt'
+import { useDispatch } from 'react-redux'
+import { useAktivitetspliktOppgaveVurdering } from '~components/aktivitetsplikt/OppgaveVurderingRoute'
+import { isPending, isSuccess, mapFailure, mapSuccess } from '~shared/api/apiUtils'
+import { ferdigstillOppgave } from '~shared/api/oppgaver'
+import { ApiErrorAlert } from '~ErrorBoundary'
+
+export function FerdigstillAktivitetspliktBrevModal() {
+  return <FerdigstillModal type="BREV" key="BREV" />
+}
+
+export function FerdigstillAktivitetspliktOppgaveModal() {
+  return <FerdigstillModal type="OPPGAVE" key="OPPGAVE" />
+}
+
+function FerdigstillModal(props: { type: 'BREV' | 'OPPGAVE' }) {
+  const { type } = props
+  const { oppgave } = useAktivitetspliktOppgaveVurdering()
+  const [open, setOpen] = useState(false)
+  const [ferdigstillBrevOgOppgaveStatus, ferdigstillOppgaveOgBrev] = useApiCall(ferdigstillBrevOgOppgaveAktivitetsplikt)
+  const [ferdigstillOppgaveStatus, apiFerdigstillOppgave] = useApiCall(ferdigstillOppgave)
+
+  const dispatch = useDispatch()
+  const ferdigstill = () => {
+    if (type === 'OPPGAVE') {
+      apiFerdigstillOppgave(oppgave.id, (oppgave) => {
+        dispatch(setAktivitetspliktOppgave(oppgave))
+      })
+    } else if (type === 'BREV') {
+      ferdigstillOppgaveOgBrev({ oppgaveId: oppgave.id }, (oppgave) => {
+        dispatch(setAktivitetspliktOppgave(oppgave))
+      })
+    }
+  }
+
+  return (
+    <>
+      <Button variant="primary" onClick={() => setOpen(true)}>
```

Comment:
Jeg er nesten litt der at dette kunne vært to komponenter? Ser bra ut ellers!

Reply:
Ja, jeg kan se om det blir rimeligere